java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/memory/azurecognitivesearch/AzureCognitiveSearchMemory.java

```diff
@@ -0,0 +1,362 @@
+                                                        new SemanticConfiguration(
+                                                                "default",
+                                                                new PrioritizedFields()
+                                                                        .setTitleField(
+                                                                                new SemanticField()
+                                                                                        .setFieldName(
+                                                                                                "Description"))
+                                                                        .setPrioritizedContentFields(
+                                                                                Arrays.asList(
+                                                                                        new SemanticField()
+                                                                                                .setFieldName(
+                                                                                                        "Text"),
+                                                                                        new SemanticField()
+                                                                                                .setFieldName(
+                                                                                                        "AdditionalMetadata")))))));
+        return this._adminClient.createIndex(newIndex);
+    }
+
+    @SuppressWarnings("unchecked")
+    private Mono<String> upsertRecordAsync(String indexName, AzureCognitiveSearchRecord record) {
+        SearchAsyncClient client = this.getSearchClient(indexName);
+        IndexDocumentsOptions throwOnAnyError =
+                new IndexDocumentsOptions().setThrowOnAnyError(true);
+
+        Function<Response<IndexDocumentsResult>, String> transform =
+                (response) -> {
+                    if (response.getStatusCode() == 404
+                            || response.getValue() == null
+                            || response.getValue().getResults() == null
+                            || response.getValue().getResults().isEmpty()) {
+                        throw new AzureCognitiveSearchMemoryException(
+                                "Memory write returned null or an empty set");
+                    }
+                    return response.getValue().getResults().get(0).getKey();
+                };
+
+        return Mono.fromCallable(
+                () -> {
+                    try {
+                        return client.mergeOrUploadDocumentsWithResponse(
+                                        Collections.singleton(record), throwOnAnyError)
+                                .map(transform)
+                                .block();
```

Comment:
should not need to block, can just return the mono without wrapping in the fromCallable

Reply:
Replaced with 'onErrorResume'